### DISCUSSION TITLE
Pulumi upgrade - "File too large"

### DISCUSSION THREAD

**Discussion by n2taylor** [Homebrew/discussions#487] [Everyday usage]
Seeing a "file too large" error when the Pulumi source build tries to download 
```
gopkg.in/src-d/go-git-fixtures.v3@v3.5.0: write /Users/nick/Library/Caches/Homebrew/go_mod_cache/pkg/mod/cache/download/gopkg.in/src-d/go-git-fixtures.v3/@v/v3.5.0.zip.tmp011137913: file too large
```

This happens on `brew upgrade pulumi`, and it builds from source because the bottle installation fails on: `Error: Failed to download resource "pulumi"`.

In general, since I upgraded to Big Sur (not Apple Silicon) I'm seeing a lot of issues both with downloading bottles and source builds failing either in the Go module download step or the tarball extraction step.

**Comment by gromgit** [comment]
What's the output of `ulimit -a`?

**Reply by n2taylor** [reply]
```
-t: cpu time (seconds)              unlimited
-f: file size (blocks)              65536
-d: data seg size (kbytes)          unlimited
-s: stack size (kbytes)             8192
-c: core file size (blocks)         0
-v: address space (kbytes)          unlimited
-l: locked-in-memory size (kbytes)  unlimited
-u: processes                       2784
-n: file descriptors                65536
```

**Reply by n2taylor** [reply]
`ulimit -f unlimited` seems to fix it. Thanks... wasn't expecting that value to change over an OS upgrade...

**Reply by gromgit** [reply]
> -f: file size (blocks)              65536

That means you can only write files up to 64MB, which almost certainly explains the problems you've been having.
```
ulimit -f unlimited
```
should fix that for the life of your shell session, but you'll have to find and fix the point at which that low file limit is being set, otherwise it'll bite you each time you log in again.
